src/Umbraco.Core/Models/ContentBase.cs

```diff
@@ -278,7 +273,8 @@ public virtual void SetValue(string propertyTypeAlias, object value, string cult
                 return;
             }
 
-            var propertyType = PropertyTypes.FirstOrDefault(x => x.Alias.InvariantEquals(propertyTypeAlias));
+            var propertyTypes = Current.Services.ContentTypeBaseServices.GetContentTypeOf(this).CompositionPropertyTypes;
```

Comment:
This one is particularly ugly :( Can we pass in this service to this method? Or does this method need to belong to this class (i.e. could be an ext method)